Makefile

```diff
@@ -36,7 +40,7 @@ $(BUILD_DIR)/configure: $(BUILD_DIR)/config.m4
 $(BUILD_DIR)/Makefile: $(BUILD_DIR)/configure
 	$(Q) (cd $(BUILD_DIR); ./configure)
 
-$(SO_FILE): $(ALL_FILES) $(BUILD_DIR)/Makefile
+$(SO_FILE): $(C_FILES) $(BUILD_DIR)/Makefile
```

Comment:
Note that `$(BUILD_DIR)/config.m4 $(M4_FILES)` is not included as a dependency as they are already transitive dependencies of `$(BUILD_DIR)/Makefile`